backend.go

```diff
@@ -45,8 +46,9 @@ func findAllBackends(mainLoose *fileStorer, mainPacked *pack.Storage, root strin
 
 	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
-		storage = append(storage, newFileStorer(scanner.Text(), ""))
-		pack, err := pack.NewStorage(scanner.Text())
+		alternateRoot := filepath.Join(root, scanner.Text())
```

Comment:
This should work fine if the path in the alternates file is relative, but I don't think that's usually the case. It looks like Git usually writes absolute paths into this file, and this change would break alternates here.

I think we'd need a solution that works both for relative and absolute paths.

Reply:
Tks for reply, I lost the caes that alternates may be absolute paths, I will change the commits latter.